src/operator/numpy/np_unique_op.cc

```diff
@@ -0,0 +1,326 @@
+    });
+    int output_flag = 0;
+    if (param.return_index) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_1);
+      Tensor<cpu, 1, dim_t> outdata =
+          outputs[output_flag].data().get_with_shape<cpu, 1, dim_t>(Shape1(1), s);
+      ASSIGN_DISPATCH(outdata, OpReqType::kWriteTo, 0);
+    }
+    if (param.return_inverse) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_1);
+      Tensor<cpu, 1, dim_t> outdata =
+          outputs[output_flag].data().get_with_shape<cpu, 1, dim_t>(Shape1(1), s);
+      ASSIGN_DISPATCH(outdata, OpReqType::kWriteTo, 0);
+    }
+    if (param.return_counts) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_1);
+      Tensor<cpu, 1, dim_t> outdata =
+          outputs[output_flag].data().get_with_shape<cpu, 1, dim_t>(Shape1(1), s);
+      ASSIGN_DISPATCH(outdata, OpReqType::kWriteTo, 1);
+    }
+  } else if (inputs[0].shape().Size() == 0) {
+    // If the input tensor is zero size, only a check on axis is needed
+    if (param.axis.has_value()) {
+      int axis = param.axis.value();
+      if (axis < 0) axis += inputs[0].shape().ndim();
+      CHECK(axis >= 0 && axis < inputs[0].shape().ndim())
+        << "Axis must be within the range of input tensor's dimension";
+    }
+    // set the shapes of outputs
+    mxnet::TShape shape_0(1, 0);
+    const_cast<NDArray &>(outputs[0]).Init(shape_0);
+    int output_flag = 0;
+    if (param.return_index) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_0);
+    }
+    if (param.return_inverse) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_0);
+    }
+    if (param.return_counts) {
+      output_flag += 1;
+      const_cast<NDArray &>(outputs[output_flag]).Init(shape_0);
+    }
+  } else {
+    if (!param.axis.has_value()) {
+      NumpyUniqueCPUNoneAxisImpl(param, ctx, inputs, req, outputs);
+    } else {
+      NumpyUniqueCPUImpl(param, ctx, inputs, req, outputs);
+    }
+  }
+}
+
+DMLC_REGISTER_PARAMETER(NumpyUniqueParam);
+
+NNVM_REGISTER_OP(_npi_unique)
+.set_attr_parser(ParamParser<NumpyUniqueParam>)
+.set_num_inputs(1)
+.set_num_outputs([](const NodeAttrs& attrs) {
+    const NumpyUniqueParam& param = nnvm::get<NumpyUniqueParam>(attrs.parsed);
+    int output_num = 1;
+    if (param.return_index) output_num += 1;
+    if (param.return_inverse) output_num += 1;
+    if (param.return_counts) output_num += 1;
+    return output_num;
+  })
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+  [](const NodeAttrs& attrs) {
+    return std::vector<std::string>{"data"};
+  })
+.set_attr<nnvm::FInferType>("FInferType", NumpyUniqueType)
+.set_attr<FComputeEx>("FComputeEx<cpu>", NumpyUniqueCPUForward)
+.set_attr<FInferStorageType>("FInferStorageType", NumpyUniqueStorageType)
+.set_attr<nnvm::FGradient>("FGradient", MakeZeroGradNodes)
```

Comment:
Do not register this since gradient of this op should not be zero. If you want to circumvent the limitation of HybridBlock not taking non-differentiable ops, register a dummy backward function with clear error message saying that "back-propagation is not supported for this op".